packages/mobile/src/transactions/feed/TransactionFeed.tsx

```diff
@@ -131,11 +93,106 @@ function mapStandbyTransactionToFeedTokenTransaction(tx: StandbyTransaction): Fe
+  setFetchedTransactions: any
+) {
+  return (transactions: TokenTransaction[]) => {
+    const currentHashes = new Set(fetchedTransactions.map((tx) => tx.transactionHash))
+
+    const transactionsWithoutDuplicatedHash = fetchedTransactions.concat(
+      transactions.filter((tx) => !currentHashes.has(tx.transactionHash))
+    )
+
+    transactionsWithoutDuplicatedHash.sort((a, b) => {
+      return b.timestamp - a.timestamp
+    })
+
+    setFetchedTransactions(transactionsWithoutDuplicatedHash)
+  }
+}
+
 function TransactionFeed() {
+  const address = useSelector(walletAddressSelector)
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+
+  const [pageInfo, setPageInfo] = useState<PageInfo | null>(null)
+  const [fetchMoreTransactions, setFetchMoreTransactions] = useState(false)
+
   const cachedTransactions = useSelector(transactionsSelector)
+  const [fetchedTransactions, setFetchedTransactions] = useState<TokenTransaction[]>([])
+  const addFetchedTransactions = addFetchedTransactionFunction(
+    fetchedTransactions,
+    setFetchedTransactions
+  )
+
+  const dispatch = useDispatch()
+
+  // Update the counter variable every |POLL_INTERVAL| so that a query is made to the backend.
+  const [counter, setCounter] = useState(0)
+  useInterval(() => setCounter((n) => n + 1), POLL_INTERVAL)
+
+  const { loading, error } = useAsyncQueryTransactionsFeed({
+    address,
+    localCurrencyCode,
+    dependencies: [counter],
+    onSuccess: (result) => {
+      if (result?.data?.tokenTransactionsV2?.transactions.length) {
+        addFetchedTransactions(result.data.tokenTransactionsV2.transactions)
+        dispatch(updateTransactions(result.data.tokenTransactionsV2.transactions))
+      }
+      if (!pageInfo) {
+        setPageInfo(result?.data?.tokenTransactionsV2?.pageInfo)
+      }
+      if (result?.errors) {
+        Sentry.captureException(result.errors)
+        Logger.warn(
+          TAG,
+          ` Found errors when querying for new transactions in feed: ${JSON.stringify(
+            result.errors
+          )}`
+        )
+      }
+    },
+    onError: (error) => {
+      Sentry.captureException(error)
+      Logger.warn(TAG, ` Found errors when querying for new transactions in feed: ${error.message}`)
+    },
+  })
 
-  const { loading, error, transactions } = useQueryTransactionFeed()
-  const confirmedTokenTransactions: TokenTransaction[] = transactions ?? cachedTransactions
+  useAsyncQueryTransactionsFeed({
+    address,
+    localCurrencyCode,
+    afterCursor: pageInfo?.endCursor,
+    dependencies: [counter, fetchMoreTransactions, pageInfo?.endCursor],
+    precondition: () => {
+      return fetchMoreTransactions && !!pageInfo && pageInfo?.hasNextPage
+    },
+    onSuccess: (result) => {
+      // TODO: Show a message if we know there aren't more transactions
+      setFetchMoreTransactions(false)
+
+      if (result) {
+        if (result?.data?.tokenTransactionsV2?.transactions.length) {
```

Comment:
nit: don't need the `?` after `result` since it's already inside the if statement. what do you think about using `if (result?.data)` instead as the if condition? `result` should always exist for graphql right?